src/production/docs/codesplitting.md

```diff
@@ -0,0 +1,61 @@
+# Code splitting in production
+The desired structure after CodeSplittingPhase of the productionContext
+
+## Example source code
+
+```js
+index.ts
+  import a;
+  import b;
+
+a.ts
+  await import("./foo")
+
+b.ts
+  await import("./foo")
+
+foo.ts:
+   import "./bar"
+
+bar.ts:
+    ...
+```
+
+```js
+IProductionContext: {
+  ctx: Context,
+  modules: Array<Module>,
+  splitEntries: SplitEntries<{
+    entries: Array<SplitEntry?>,
+    register: function() {}
+  }>
+}
+```
+
+```js
+ModuleTypes: {
+  SPLIT_MODULE,
+  MAIN_MODULE
+}
+```
+
+```js
+splitEntries: {
+  entries: [{
+    entry: { // FooModule:
+      moduleTree: {
+        ...,
+        moduleType: SPLIT_MODULE
```

Comment:
seems overkill if a module is in the splitEntries... thinking about removing the moduleType here..
Or do you need it in a later process? @nchanged

Reply:
I think it's needed yes

Reply:
Leaving it alone then ;)